pkg/controller/nodeipam/ipam/range_allocator.go

```diff
@@ -253,41 +303,38 @@ func (r *rangeAllocator) occupyCIDRs(node *v1.Node) error {
 // function you have to make sure to update nodesInProcessing properly with the
 // disposition of the node when the work is done.
 func (r *rangeAllocator) AllocateOrOccupyCIDR(logger klog.Logger, node *v1.Node) error {
+	r.lock.Lock()
+	defer r.lock.Unlock()
```

Comment:
I can't remember why do we need to lock here?

Reply:
After tracing the history of the code, I think this lock was only used for ‘nodesInProcessing’, which has been deleted in this PR. Therefore, I think it’s okay to remove it.